blog/26-04-2023-building-a-celo-based-decentralized-medical-record-system/building-a-celo-based-decentralized-medical-record-system.md

```diff
@@ -0,0 +1,833 @@
+      setError(error.message);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+  const updateExistingRecord = async (e) => {
+    e.preventDefault();
+    const tokenId = parseInt(selectedRecordId);
+    setIsLoading(true);
+    setError(null);
+    try {
+      await contract.methods.updateRecord(tokenId, medicalData).send({ from: address });
+      const updatedRecords = records.map((record, index) =>
+        index + 1 === tokenId ? { ...record, medicalData } : record
+      );
+      setRecords(updatedRecords);
+      setMedicalData("");
+      setSelectedRecordId(null);
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+  const MedicalRecordList = ({ records, onSelect }) => {
+    return (
+      <ul>
+        {records.map((record, index) => (
+          <li key={index} onClick={() => onSelect(index + 1)}>
+            {record.name} - {record.birthdate}
+          </li>
+        ))}
+      </ul>
+    );
+  };
+  return (
+    <div>
+      {!isConnected && (
+        <button onClick={handleConnect}>Connect Wallet</button>
+      )}
+      {isConnected && (
+        <>
+          <h1>My Medical Records</h1>
+          <form onSubmit={selectedRecordId ? updateExistingRecord : createNewRecord}>
+            <div>
+              <label>
+                Name:
+                <input
+                  type="text"
+                  value={name}
+                  onChange={(e) => setName(e.target.value)}
+                  disabled={!!selectedRecordId}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                Birthdate:
+                <input
+                  type="date"
+                  value={birthdate}
+                  onChange={(e) => setBirthdate(e.target.value)}
+                  disabled={!!selectedRecordId}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                Medical Data:
+                <textarea
+                  value={medicalData}
+                  onChange={(e) => setMedicalData(e.target.value)}
+                />
+              </label>
+            </div>
+            {selectedRecordId ? (
+              <div>
+                <button type="submit">Update Record</button>
+                <button type="button" onClick={() => setSelectedRecordId(null)}>
+                  Cancel
+                </button>
+              </div>
+            ) : (
+              <button type="submit">Create Record</button>
+            )}
+          </form>
+          <div>
+            <h2>Your Medical Records</h2>
+            <MedicalRecordList records={records} onSelect={setSelectedRecordId} />
+          </div>
+          {error && <p>Error: {error}</p>}
+          {isLoading && <p>Loading...</p>}
+        </>
+      )}
+    </div>
+  );
+};
+export default MedicalRecord;
+```
+
+On running the command again:
+```bash
```

Comment:
can you add a space before the snippet here

Reply:
done